### PR TITLE
Add the code for Jenkins continuous integration

### PR DESCRIPTION
**Pull request by ebecheva**
The code consists of:
- Jenkinsfile
- some scripts

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+                sh '''
+                uname -a
+                whoami
+                pwd
+                ls -l
+                if [ -d "./HGCTPGValidation" ] 
+                then
+                    rm -rf HGCTPGValidation
+                fi
+                git clone -b Jenkins-developments https://github.com/hgc-tpg/HGCTPGValidation HGCTPGValidation
```
Comment (jbsauvan) [inline Jenkinsfile:24]: Shouldn't it fetch the master branch?

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+                ~/grid_login
```
Comment (jbsauvan) [inline Jenkinsfile:25]: `grid_login` is called in multiple places. Is it needed to call it even in jobs where grid data are not fetched?

File: config/user_parameters_testJenkins_cfg.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    inputRefFileName = 'root://cms-xrd-global.cern.ch///eos/cms//store/mc/Phase2HLTTDRWinter20DIGI/QCD_Pt-15to7000_TuneCP5_Flat_14TeV-pythia8/GEN-SIM-DIGI-RAW/FlatPU0To200_castor_110X_mcRun4_realistic_v3_ext1-v1/260000/003F2BA9-0D02-5A43-A53F-4161E513BFA6.root',
+    inputTestFileName = 'root://cms-xrd-global.cern.ch///eos/cms//store/mc/Phase2HLTTDRWinter20DIGI/QCD_Pt-15to7000_TuneCP5_Flat_14TeV-pythia8/GEN-SIM-DIGI-RAW/FlatPU0To200_castor_110X_mcRun4_realistic_v3_ext1-v1/260000/003F2BA9-0D02-5A43-A53F-4161E513BFA6.root',
+    #customiseRefFile = 'L1Trigger/L1THGCal/customClustering.custom_2dclustering_constrainedtopological',
+    #customiseTestFile = 'L1Trigger/L1THGCal/customClustering.custom_2dclustering_constrainedtopological',
+    customiseRefFile = '',
+    customiseTestFile = '',
+    dropedBranches = '"drop l1tEMTFHit2016Extras_simEmtfDigis_CSC_HLT","drop l1tEMTFHit2016Extras_simEmtfDigis_RPC_HLT","drop l1tEMTFHit2016s_simEmtfDigis__HLT","drop l1tEMTFTrack2016Extras_simEmtfDigis__HLT","drop l1tEMTFTrack2016s_simEmtfDigis__HLT"',
+    webDirPath = './GIFS'
+    )
```
Comment (jbsauvan) [inline config/user_parameters_testJenkins_cfg.py:49]: Is this file used?

File: Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+            stages{
+                stage('Install'){
+                    steps {
+                        echo 'InstallCMSSW Test step..'
+                        sh '''
+                        pwd
+                        ~/grid_login
+                        cd test_dir
+                        source ../HGCTPGValidation/scripts/extractReleaseName.sh $CHANGE_TARGET
+                        unset IFS
```
Comment (jbsauvan) [inline Jenkinsfile:48]: Here and below: what is the `IFS` variable?

File: scripts/produceData.sh [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+if [[ -n $2 ]] #if not empty
+then
+  cmsDriver.py hgcal_tpg_validation -n 50 + \
+  --mc --eventcontent FEVTDEBUG --datatier GEN-SIM-DIGI-RAW + \
+  --conditions auto:phase2_realistic + \
+  --beamspot HLLHC14TeV + \
+  --step USER:Validation/HGCalValidation/hgcalRunEmulatorValidationTPG_cff.hgcalTPGRunEmulatorValidation + \
+  --geometry Extended2026D49 --era Phase2C9 --procModifiers $2 + \
+  --inputCommands "keep *","drop l1tEMTFHit2016Extras_simEmtfDigis_CSC_HLT","drop l1tEMTFHit2016Extras_simEmtfDigis_RPC_HLT","drop l1tEMTFHit2016s_simEmtfDigis__HLT","drop l1tEMTFTrack2016Extras_simEmtfDigis__HLT","drop l1tEMTFTrack2016s_simEmtfDigis__HLT" + \
+  --filein file:/data_CMS_upgrade/becheva/data_jenkins/003F2BA9-0D02-5A43-A53F-4161E513BFA6.root + \
```
Comment (jbsauvan) [inline scripts/produceData.sh:23]: - Would be better to put the data in a "non-user" directory
- Need to keep track of the dataset name. At the moment it is not clear what these data are.

File: scripts/produceData.sh [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  --customise_commands "process.MessageLogger.files.out_$1 = dict(); process.Timing = cms.Service('Timing', summaryOnly = cms.untracked.bool(False), useJobReport = cms.untracked.bool(True)); process.SimpleMemoryCheck = cms.Service('SimpleMemoryCheck', ignoreTotal = cms.untracked.int32(1)); process.schedule = cms.Schedule(process.user_step)"
+else
+  cmsDriver.py hgcal_tpg_validation -n 50 + \
+  --mc --eventcontent FEVTDEBUG --datatier GEN-SIM-DIGI-RAW + \
+  --conditions auto:phase2_realistic + \
+  --beamspot HLLHC14TeV + \
+  --step USER:Validation/HGCalValidation/hgcalRunEmulatorValidationTPG_cff.hgcalTPGRunEmulatorValidation + \
+  --geometry Extended2026D49 --era Phase2C9 + \
+  --inputCommands "keep *","drop l1tEMTFHit2016Extras_simEmtfDigis_CSC_HLT","drop l1tEMTFHit2016Extras_simEmtfDigis_RPC_HLT","drop l1tEMTFHit2016s_simEmtfDigis__HLT","drop l1tEMTFTrack2016Extras_simEmtfDigis__HLT","drop l1tEMTFTrack2016s_simEmtfDigis__HLT" + \
+  --filein file:/data_CMS_upgrade/becheva/data_jenkins/003F2BA9-0D02-5A43-A53F-4161E513BFA6.root + \
```
Comment (jbsauvan) [inline scripts/produceData.sh:34]: Same as before

File: scripts/writeToFile.py [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+def writeIntoFile(prnumber, prtitle):
+    with open('validation_webpages.txt', 'w') as f:
+        f.write(prtitle)
+    
+def main(dirname, prnumber, prtitle):
+    print('dirname = ', dirname)
+    print('prtitle = ', prtitle)
+    print('prnumber= ', prnumber)
+    #os.mkdir(dirname)
```
Comment (jbsauvan) [inline scripts/writeToFile.py:15]: Commented code to be removed

File: scripts/installCMSSW.sh [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+echo $relversion
+remote=$3
+echo $remote
+branch=$4
+echo $branch
+branch_ref=$5
+echo $branch_ref
+label=$6
+echo $label
+#echo "directory = " ${relversion}_HGCalTPGValidation_$label
```
Comment (jbsauvan) [inline scripts/installCMSSW.sh:25]: Commented code to be removed

File: Jenkinsfile (hunk 2 above)
Comment (ebecheva) [reply Jenkinsfile:25]: The grid_login was added for accessing grid data. If we plan to use only local data files, it is not needed.

File: Jenkinsfile (hunk 1 above)
Comment (ebecheva) [reply Jenkinsfile]: Yes, it should be the master branch.

File: scripts/produceData.sh (hunk 5 above)
Comment (ebecheva) [reply scripts/produceData.sh:23]: Yes, it is better to use "non-user" directory. I will move the data file to /data_CMS_upgrade/data_jenkins.

File: scripts/produceData.sh (hunk 6 above)
Comment (ebecheva) [reply scripts/produceData.sh:34]: OK

File: scripts/writeToFile.py (hunk 7 above)
Comment (ebecheva) [reply scripts/writeToFile.py]: Done

File: scripts/installCMSSW.sh (hunk 8 above)
Comment (ebecheva) [reply scripts/installCMSSW.sh]: Removed

File: config/user_parameters_testJenkins_cfg.py (hunk 3 above)
Comment (ebecheva) [reply config/user_parameters_testJenkins_cfg.py:49]: This configuration file was used in the very first version of the Jenkins pipeline. It is not used anymore.

File: scripts/produceData.sh (hunk 5 above)
Comment (jbsauvan) [reply scripts/produceData.sh]: Thanks. Could it be moved in a subdirectory made of the dataset name?

Comment (ebecheva) [reply scripts/produceData.sh]: It was already done yesterday

**Pull request merged**